File: _Season15/01 - Not Quite Lisp/Python/aoc2015_01_1.py
"""
aoc2015_01_1.py
----------------
Author: Nida Anis
Date created: 20/02/2025
----------------
Description:
- Advent of Code 2015 Day 1: Not Quite Lisp
- Solution to Part 1 (Part 2 in aoc2015_01_2.py)
"""

def get_map_from_file():
    """Reads the map from the input file."""
    map = []

    with open("input.txt", "r") as f:
        while 1:
            char = f.read(1)
            map.append(char)
            if not char:
                break
    
    return map

def get_final_floor_number(map):
    """Gets Santa's final floor number."""
    floor_num = 0

    for char in map:
        if char == "(":
            floor_num += 1

        elif char == ")":
            floor_num -= 1
    
    return floor_num

def main():
    map = get_map_from_file()
    floor_num = get_final_floor_number(map)
    print(f"The instructions take Santa to Floor {floor_num}.")

if __name__ == "__main__":
    main()

File: _Season15/01 - Not Quite Lisp/C#/NotQuiteLisp/aoc2015_01_1.cs
/*
aoc2015_01_1.cs
----------------
Author: Nida Anis
Date: 31/12/2024
----------------
Description:
- Advent of Code 2015 Day 1: Not Quite Lisp
- Solution to Part 1 (Part 2 in aoc2015_01_2.cs)
*/

class Part1
{

    /// <summary>
    /// Gets Santa's final floor number.
    /// </summary>
    /// <param name="charArray"></param>
    /// <returns>Santa's final floor number.</returns>
    public static int GetFinalFloorNumber(char[] charArray)
    {
        int floorNumber = 0;

        for (int i = 0; i < charArray.Length; i++)
        {
            if (charArray[i] == '(')
            {
                floorNumber += 1;
            }

            else if (charArray[i] == ')')
            {
                floorNumber -= 1;
            }
        }

        return floorNumber;
    }
}